src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java

```diff
@@ -266,7 +267,12 @@ public void setQuery(String query) {
     @Override
     public void beginTree(DetailAST rootAST) {
         if (xpathExpression != null) {
-            final List<DetailAST> matchingNodes = findMatchingNodesByXpathQuery(rootAST);
+            DetailAST astWithComments = rootAST;
+            // This check is being done so we do not add comments when it is not being checked
+            if (query.contains("COMMENT")) {
+                astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);
+            }
```

Comment:
this is a very weird way.
You can just add 
```
@Override
    public boolean isCommentNodesRequired() {
        return true;
    }
```
to class and it will work with comment nodes as well.

Reply:
Thanks! I was not aware of this method.